python/tank/util/filesystem.py

```diff
@@ -343,3 +343,64 @@ def create_valid_filename(value):
         # so that we return a string
         u_src = value.decode("utf-8")
         return exp.sub("_", u_src).encode("utf-8")
+
+def get_permissions(path):
+    """
+    Retrieve the file system permissions for the file or folder in the
+    given path.
+
+    :param filename: Path to the file to be queried for permissions
+    :returns: permissions bits of the file 
+    :raises: OSError - if there was a problem retrieving permissions for the path
+    """
+    return stat.S_IMODE(os.lstat(path)[stat.ST_MODE])
+
+def delete_folder(path):
+    """
+    Deletes a folder and all of its contents recursively, even if it has read only items
+
+    :param path: File system path to location to the folder to be deleted
+    :returns: True if the folder and all of its contents were deleted without
+              any contingencies. False if there were any issues removing
+              any items in the folder. Problems deleting any items will be
+              reported as errors in the log output but otherwise ignored and
+              skipped; meaning the function will continue deleting as much
+              as it can.
+    """
+    deleted = True
+
+    def _on_rm_error(func, path, exc_info):
+        # On Windows, Python's shutil can't delete read-only files, so if we were trying to delete one,
+        # remove the flag.
+        # Inspired by http://stackoverflow.com/a/4829285/1074536
+        global deleted
+        if func == os.unlink or func == os.remove or func == os.rmdir:
+            try:
+                attr = os.stat(path)[stat.ST_MODE]
+                if not (attr & stat.S_IWRITE):
+                    os.chmod(path, stat.S_IWRITE | attr)
+                    try:
+                        func(path)
+                    except Exception, e:
+                        log.error("Could not delete %s: %s. Skipping" % (path, e))
+                        deleted = False
+                else:
+                    log.error("Could not delete %s: %s. Skipping" % (path, e))
+                    deleted = False
+            except Exception, e:
+                log.error("Could not delete %s: %s. Skipping" % (path, e))
+                deleted = False
+        else:
+            log.error("Could not delete %s. Skipping." % path)
+            deleted = False
+
+    if os.path.exists(path):
+        try:
+            shutil.rmtree(path, onerror=_on_rm_error)
+        except Exception, e:
+            log.error("Could not delete %s: %s" % (path, e))
```

Comment:
i suggest demoting these errors to warnings. Error means something bad happened, we don't know for sure that the failed deletion is actually bad - (in the case of a backup for example, it's not bad)